troposphere/sqs.py

```diff
@@ -34,6 +34,12 @@ class Queue(AWSObject):
         'VisibilityTimeout': (integer, False),
     }
 
+    def validate(self):
+        if self.properties.get('FifoQueue'):
+            if not self.properties.get('QueueName', '').endswith('.fifo'):
```

Comment:
The [docs](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html#aws-sqs-queue-name) indicate QueueName is not required. But this code would raise an error if FifoQueue was specified without QueueName.